addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -855,8 +855,9 @@ private boolean performElementAttack(HtmlContext context, HttpMessage msg, Strin
         // In this case the parent effectively changes
         List<HtmlContext> context2 =
                 performAttack(msg, param, attackString1, context, HtmlContext.IGNORE_PARENT);
-        if (context2 == null) {
```

Comment:
Sonar actually warns that this can never be true.

Reply:
Raise a bug.